src/geom/stroke.rs

```diff
@@ -59,15 +146,124 @@ impl<T: Debug> Debug for PartialStroke<T> {
     }
 }
 
+cast_from_value! {
+    LineCap: "line cap",
+    "butt" => Self::Butt,
+    "round" => Self::Round,
+    "square" => Self::Square,
+}
+
+cast_from_value! {
+    LineJoin: "line join",
+    "miter" => Self::Miter,
+    "round" => Self::Round,
+    "bevel" => Self::Bevel,
+}
+
+cast_from_value! {
+    DashLength: "dash length",
+    "dot" => Self::LineWidth,
+    l: Length => Self::Length(l),
+}
+
+impl Resolve for DashLength {
+    type Output = DashLength<Abs>;
+
+    fn resolve(self, styles: StyleChain) -> Self::Output {
+        match self {
+            Self::LineWidth => DashLength::LineWidth,
+            Self::Length(l) => DashLength::Length(l.resolve(styles)),
+        }
+    }
+}
+
+cast_from_value! {
+    DashPattern: "dash pattern",
+    // Use same names as tikz:
+    // https://tex.stackexchange.com/questions/45275/tikz-get-values-for-predefined-dash-patterns
+    "solid" => Vec::new().into(),
+    "dotted" => vec![DashLength::LineWidth, Abs::pt(2.0).into()].into(),
+    "densely dotted" => vec![DashLength::LineWidth, Abs::pt(1.0).into()].into(),
+    "loosely dotted" => vec![DashLength::LineWidth, Abs::pt(4.0).into()].into(),
+    "dashed" => vec![Abs::pt(3.0).into(), Abs::pt(3.0).into()].into(),
+    "densely dashed" => vec![Abs::pt(3.0).into(), Abs::pt(2.0).into()].into(),
+    "loosely dashed" => vec![Abs::pt(3.0).into(), Abs::pt(6.0).into()].into(),
+    "dashdotted" => vec![Abs::pt(3.0).into(), Abs::pt(2.0).into(), DashLength::LineWidth, Abs::pt(2.0).into()].into(),
+    "densely dashdotted" => vec![Abs::pt(3.0).into(), Abs::pt(1.0).into(), DashLength::LineWidth, Abs::pt(1.0).into()].into(),
+    "loosely dashdotted" => vec![Abs::pt(3.0).into(), Abs::pt(4.0).into(), DashLength::LineWidth, Abs::pt(4.0).into()].into(),
```

Comment:
not a huge fan of these names, but if we keep them, they should use kebab-case, e.g. `"loosely-dash-dotted"`.

Reply:
I don't have strong opinions here other than that I think we should predefine a couple of useful dash patterns. These names seem to have some precedence though as they are also used in matplotlib: https://matplotlib.org/stable/gallery/lines_bars_and_markers/linestyles.html